kubernetes.go

```diff
@@ -44,32 +62,71 @@ type Kubernetes struct {
 	PersistentVolumeClaims *persistentvolumeclaims.PersistentVolumeClaims
 }
 
-type KubernetesOptions struct {
+// KubeConfig represents the initialization settings for the kubernetes api client.
+type KubeConfig struct {
 	ConfigPath string
 }
 
-func (obj *Kubernetes) XKubernetes(ctx *context.Context, options KubernetesOptions) (*Kubernetes, error) {
+// Ensure the interfaces are implemented correctly.
+var (
+	_ modules.Module   = &RootModule{}
+	_ modules.Instance = &ModuleInstance{}
+)
+
+// NewModuleInstance implements the modules.Module interface to return
+// a new instance for each VU.
+func (*RootModule) NewModuleInstance(vu modules.VU) modules.Instance {
+	mi := &ModuleInstance{
+		Version: version,
+		vu:      vu,
+		exports: make(map[string]interface{}),
+	}
+	mi.exports["Kubernetes"] = mi.newClient
+
+	return mi
+}
+
+// Exports implements the modules.Instance interface and returns the exports
+// of the JS module.
+func (mi *ModuleInstance) Exports() modules.Exports {
+	return modules.Exports{
+		Named: mi.exports,
```

Comment:
nit: I really do think you can just create the object here instead of adding it to the ModuleInstance